nautilus_core/infrastructure/src/sql/models/orders.rs

```diff
@@ -0,0 +1,356 @@
+//  Copyright (C) 2015-2024 Nautech Systems Pty Ltd. All rights reserved.
+//  https://nautechsystems.io
+//
+//  Licensed under the GNU Lesser General Public License Version 3.0 (the "License");
+//  You may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at https://www.gnu.org/licenses/lgpl-3.0.en.html
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------------------------------
+
+use std::{collections::HashMap, str::FromStr};
+
+use nautilus_core::{nanos::UnixNanos, uuid::UUID4};
+use nautilus_model::{
+    enums::{ContingencyType, OrderSide, OrderType, TimeInForce, TrailingOffsetType, TriggerType},
+    events::order::{
+        accepted::OrderAccepted, cancel_rejected::OrderCancelRejected, canceled::OrderCanceled,
+        denied::OrderDenied, emulated::OrderEmulated, event::OrderEventAny, expired::OrderExpired,
+        filled::OrderFilled, initialized::OrderInitialized, modify_rejected::OrderModifyRejected,
+        pending_cancel::OrderPendingCancel, pending_update::OrderPendingUpdate,
+        rejected::OrderRejected, released::OrderReleased, submitted::OrderSubmitted,
+        triggered::OrderTriggered, updated::OrderUpdated,
+    },
+    identifiers::{
+        client_order_id::ClientOrderId, exec_algorithm_id::ExecAlgorithmId,
+        instrument_id::InstrumentId, order_list_id::OrderListId, strategy_id::StrategyId,
+        trader_id::TraderId,
+    },
+    types::{price::Price, quantity::Quantity},
+};
+use sqlx::{postgres::PgRow, FromRow, Row};
+use ustr::Ustr;
+
+pub struct OrderEventAnyModel(pub OrderEventAny);
+pub struct OrderAcceptedModel(pub OrderAccepted);
+pub struct OrderCancelRejectedModel(pub OrderCancelRejected);
+pub struct OrderCanceledModel(pub OrderCanceled);
+pub struct OrderDeniedModel(pub OrderDenied);
+pub struct OrderEmulatedModel(pub OrderEmulated);
+pub struct OrderExpiredModel(pub OrderExpired);
+pub struct OrderFilledModel(pub OrderFilled);
+pub struct OrderInitializedModel(pub OrderInitialized);
+pub struct OrderModifyRejectedModel(pub OrderModifyRejected);
+pub struct OrderPendingCancelModel(pub OrderPendingCancel);
+pub struct OrderPendingUpdateModel(pub OrderPendingUpdate);
+pub struct OrderRejectedModel(pub OrderRejected);
+pub struct OrderReleasedModel(pub OrderReleased);
+pub struct OrderSubmittedModel(pub OrderSubmitted);
+pub struct OrderTriggeredModel(pub OrderTriggered);
+pub struct OrderUpdatedModel(pub OrderUpdated);
+
+impl<'r> FromRow<'r, PgRow> for OrderEventAnyModel {
+    fn from_row(row: &'r PgRow) -> Result<Self, sqlx::Error> {
+        let kind = row.get::<String, _>("kind");
```

Comment:
This is fine for the PR, but there might be a better way to know the `kind` and if we must use strings prefer the `stringify!` macro.